src/mixins/object.svg_export.js

```diff
@@ -180,15 +180,26 @@
     _createBaseClipPathSVGMarkup: function(objectMarkup, options) {
       options = options || {};
       var reviver = options.reviver,
-          additionalTransform = options.additionalTransform || '',
-          commonPieces = [
-            this.getSvgTransform(true, additionalTransform),
-            this.getSvgCommons(),
-          ].join(''),
-          // insert commons in the markup, style and svgCommons
-          index = objectMarkup.indexOf('COMMON_PARTS');
+        additionalTransform = options.additionalTransform || '',
+        clipPath = this.clipPath;
+      var markup = [];
+      //  first handle clip path
+      if (clipPath) {
+        clipPath.clipPathId = 'CLIPPATH_' + fabric.Object.__uid++;
+        var clipPathMarkup = '<clipPath id="' + clipPath.clipPathId + '" >\n' +
+          clipPath.toClipPathSVG(reviver) +
+          '</clipPath>\n';
+        markup.unshift(clipPathMarkup);
+      }
+      var commonPieces = [
+        this.getSvgTransform(true, additionalTransform),
+        this.getSvgCommons(),
+      ].join(''),
+        // insert commons in the markup, style and svgCommons
+        index = objectMarkup.indexOf('COMMON_PARTS');
       objectMarkup[index] = commonPieces;
-      return reviver ? reviver(objectMarkup.join('')) : objectMarkup.join('');
+      markup.push(objectMarkup.join(''));
+      return reviver ? reviver(markup.join('')) : markup.join('');
```

Comment:
ok this is a tentative to make the clipPath svg export recursive, right?
This is unrelated to brush, is a general improvement/fix and deserve its own pr with test ( testing svg export is very easy luckly )
Sorry for the extra work but i prefer we split it.

Reply:
Alright